build-core-armv7a7-imx6ull.sh

```diff
@@ -50,7 +50,7 @@ b_install $PREFIX_PROG_STRIPPED/psd-old /sbin
 #b_install "$PREFIX_PROG_STRIPPED/posixsrv" /bin
 
 b_log "Building hostutils"
-(cd phoenix-rtos-hostutils/ && make $MAKEFLAGS $CLEAN all)
+(cd phoenix-rtos-hostutils/ && make -f Makefile.old $MAKEFLAGS $CLEAN all)
```

Comment:
I would like to… but it's impossible right now because of env pollution. 
I tried to reduce it but then I found severeal issues:
1. phoenix-rtos-build bash scripts works with intention to specify variables like PREFIX_BUILD, PREFIX_A, PREFIX_H. It possibly could use defaults, that are defined in Makefile.common. Now when Makefiles are standarised I think it's advisable because there are not really pros for overwrite the defaults.
2. Pollution causes defaulting values in Makefiles. Eg. for imx6ull target: 
   1. exports are done in https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build.sh#L40-L43 CROSS is not set but exported
   2. Then phoenix-rtos-kernel is being built
 https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build-core-armv7a7-imx6ull.sh#L16
   3. CROSS is set by kernel build from common platform Makefile: https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/Makefile.arm#L9
   4. Then libphoenix is being built https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build-core-armv7a7-imx6ull.sh#L19
   5. CROSS will not be set as it's already set before because of ?=
Similar issues with other overwritable variables
3. All of this issues could be removed in the easy way by removing redundant exports but phoenix-rtos-ports are not preparing the environment for himself based on single env variable TARGET but rely on env from phoenix-rtos-build/build.sh. Also – it's not that simple how to correctly prepare env because phoenix-rtos-ports uses shell script for build 3rd party components, but env rules are defined in Makefiles.
4. For this reasons (bigger change in build script and need to revise phoenix-rtos-ports buildsystem) I decided to constantly progress by adding possibility to build components alone on host, but preserve the old Makefile for building with build.sh script.

Reply:
> I would like to… but it's impossible right now because of env pollution.
> I tried to reduce it but then I found severeal issues:

Ok, I understand there might be some issues I have hoped this PR would solve them :). If it's not possible at current time, let's merge it as-is. I'm providing some of my insights in the text below.
 
> 1. phoenix-rtos-build bash scripts works with intention to specify variables like PREFIX_BUILD, PREFIX_A, PREFIX_H. It possibly could use defaults, that are defined in Makefile.common. Now when Makefiles are standarised I think it's advisable because there are not really pros for overwrite the defaults.

Yes, some consistency in defining the ENV variables would be welcome - but that's outside this PR for  now :).

> 2. Pollution causes defaulting values in Makefiles. Eg. for imx6ull target:
>    
>    1. exports are done in https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build.sh#L40-L43 CROSS is not set but exported

It is being set in the sourced file (`build.project` sources `_targets/build.project.armv7a7-imx6ull` which have `CROSS` set).

>    2. Then phoenix-rtos-kernel is being built
>       https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build-core-armv7a7-imx6ull.sh#L16
>    3. CROSS is set by kernel build from common platform Makefile: https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/Makefile.arm#L9

The ENV cannot be polluted as this is a child process (it can't change parent's ENV).

>    4. Then libphoenix is being built https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/build-core-armv7a7-imx6ull.sh#L19
>    5. CROSS will not be set as it's already set before because of ?=
>       Similar issues with other overwritable variables
> 3. All of this issues could be removed in the easy way by removing redundant exports but phoenix-rtos-ports are not preparing the environment for himself based on single env variable TARGET but rely on env from phoenix-rtos-build/build.sh. Also – it's not that simple how to correctly prepare env because phoenix-rtos-ports uses shell script for build 3rd party components, but env rules are defined in Makefiles.
> 4. For this reasons (bigger change in build script and need to revise phoenix-rtos-ports buildsystem) I decided to constantly progress by adding possibility to build components alone on host, but preserve the old Makefile for building with build.sh script.

Let's leave the `ports` as-is for now, as they are broken for most targets :(.

Reply:
> Ok, I understand there might be some issues I have hoped this PR would solve them :). If it's not possible at current time, let's merge it as-is. I'm providing some of my insights in the text below.
I hoped as well but I think it could be bigger change.

> The ENV cannot be polluted as this is a child process (it can't change parent's ENV).
Oh, right… I strongly misunderstand things what already happened because I missed this CROSS settings in _targets/build.project.armv7a7-imx6ull… Thanks for pointing it out. But even with this, problems are the same: need to refactor phoenix-rtos-build and project and rewrite buildsystem for ports.

Reply:
ok, let's leave it for now and merge this as-is.